src/actions/VaccineActions.js

```diff
@@ -48,23 +52,27 @@ const withPermissions = async (dispatch, getState, func) => {
     return null;
   }
 
-  return func(dispatch, state);
+  return func(dispatch, getState);
 };
 
-const blinkSensor = macAddress => () => {
-  BleService().blinkWithRetries(macAddress, 3);
+const blinkSensor = macAddress => async dispatch => {
+  dispatch(blinkStart(macAddress));
+  await BleService().blinkWithRetries(macAddress, 3);
+  dispatch(blinkStop(macAddress));
 };
 
-const scanForSensors = (dispatch, state) => {
+const scanForSensors = (dispatch, getState) => {
   dispatch(scanStart());
 
   const deviceCallback = device => {
     const { id: macAddress } = device;
 
     if (macAddress) {
-      const alreadyScanned = selectScannedSensors(state);
+      const alreadyScanned = selectScannedSensors(getState());
       const alreadySaved = UIDatabase.get('Sensor', macAddress, 'macAddress');
 
+      console.log(alreadyScanned, macAddress);
```

Comment:
Looks like this might be some test code?

Reply:
cd0adf6091f4f55a1838f563f51d0304e24eab25